### PR TITLE
Add Privacy API docs

### LINKED ISSUE #47: Add docs on CMS privacy page feature
WP has a built in privacy page feature including the ability for a user to request all their data. We should document how to implement this and any special additions we add to the default behaviour in this module.

### PR DESCRIPTION
**Pull request by roborourke**
Fixes #47

### CODE REVIEW

File: docs/privacy-api.md [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+- Additional information
+  - How we protect your data
+  - What data breach procedures we have in place
+  - What third parties we receive data from
+  - What automated decision making and/or profiling we do with user data
+  - Industry regulatory disclosure requirements
+
+### Using an Existing or Alternative Page
+
+Under the Settings > Privacy admin menu item it is possible to select which page you would like to use as the privacy policy page.
```
Comment (rmccue) [inline docs/privacy-api.md]: Can we link to the settings page?

File: docs/privacy-api.md [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+
+### Extending Personal Data Export
+
+By default the export will contain any data known to be associated with the requester's email address such as comments or posts. Some code may extend the platform in such a way that it cannot determine all the data associated with the user automatically.
+
+To extend the data export use the `wp_privacy_personal_data_exporters` filter:
+
+```php
+add_filter( 'wp_privacy_personal_data_exporters', function ( $exporters ) {
+	$exporters['form-responses'] = array(
```
Comment (rmccue) [inline docs/privacy-api.md]:
```suggestion
	$exporters['form-responses'] = [
```

File: docs/privacy-api.md [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+Once they have confirmed their request a zip file is created and emailed to them.
+
+### Extending Personal Data Export
+
+By default the export will contain any data known to be associated with the requester's email address such as comments or posts. Some code may extend the platform in such a way that it cannot determine all the data associated with the user automatically.
+
+To extend the data export use the `wp_privacy_personal_data_exporters` filter:
+
+```php
+add_filter( 'wp_privacy_personal_data_exporters', function ( $exporters ) {
```
Comment (rmccue) [inline docs/privacy-api.md]:
```suggestion
add_filter( 'wp_privacy_personal_data_exporters', function ( array $exporters ) : array {
```

File: docs/privacy-api.md [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+
+		// Core group IDs include 'comments', 'posts', etc.
+		// But you can add your own group IDs as needed.
+		$group_id = 'form-responses';
+
+		// Optional group label. Core provides these for core groups.
+		// If you define your own group, the first exporter to
+		// include a label will be used as the group label in the
+		// final exported report.
+		$group_label = __( 'Form Responses' );
```
Comment (rmccue) [inline docs/privacy-api.md]: Maybe add text domains?

File: docs/privacy-api.md [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+		'meta_key' => 'email',
+		'meta_value' => $meail,
+	] );
+
+	foreach ( $responses->posts as $post ) {
+		// Most item IDs should look like postType-postID
+		// If you don't have a post, comment or other ID to work with,
+		// use a unique value to avoid having this item's export
+		// combined in the final report with other items of the same id.
+		$item_id = "form-responses-{$post->ID}";
```
Comment (rmccue) [inline docs/privacy-api.md]: `sprintf` is nicer IMO.

```suggestion
		$item_id = sprintf( 'form-responses-%d', $post->ID );
```

File: docs/privacy-api.md [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+By default the export will contain any data known to be associated with the requester's email address such as comments or posts. Some code may extend the platform in such a way that it cannot determine all the data associated with the user automatically.
+
+To extend the data export use the `wp_privacy_personal_data_exporters` filter:
+
+```php
+add_filter( 'wp_privacy_personal_data_exporters', function ( $exporters ) {
+	$exporters['form-responses'] = array(
+		'exporter_friendly_name' => __( 'Form Response Exporter' ),
+		'callback' => 'form_response_exporter',
+	);
```
Comment (rmccue) [inline docs/privacy-api.md]:
```suggestion
	];
```

File: docs/privacy-api.md [hunk 7]
```diff
@@ -0,0 +1,259 @@
...
+Should a site ever need to be restored from a backup it is important to go back through previous data removal requests and ensure the data was either already removed or to remove it again.
+
+### Extending Personal Data Erasers
+
+Again via a similar mechanism to the data export custom code can provide a means to remove personal data.
+
+A date deletion callback can be registered via the `wp_privacy_personal_data_erasers` filter:
+
+```php
+add_filter( 'wp_privacy_personal_data_erasers', function ( $erasers ) {
```
Comment (rmccue) [inline docs/privacy-api.md]:
```suggestion
add_filter( 'wp_privacy_personal_data_erasers', function ( array $erasers ) : array {
```

File: docs/privacy-api.md [hunk 8]
```diff
@@ -0,0 +1,259 @@
...
+	$erasers['form_response_eraser'] = array(
```
Comment (rmccue) [inline docs/privacy-api.md]:
```suggestion
	$erasers['form_response_eraser'] = [
```

File: docs/privacy-api.md [hunk 9]
```diff
@@ -0,0 +1,259 @@
...
+		'eraser_friendly_name' => __( 'Form Response Eraser' ),
+		'callback'             => 'form_response_eraser',
+	);
```
Comment (rmccue) [inline docs/privacy-api.md]:
```suggestion
	];
```

File: docs/privacy-api.md [hunk 10]
```diff
@@ -0,0 +1,259 @@
...
+	return $erasers;
+} );
+```
+
+The following example shows the deletion of the hypothetical `form_responses` custom post type from earlier and takes the exact same arguments of email address and page:
+
+```php
+function form_response_eraser( string $email, $page = 1 ) {
```
Comment (rmccue) [inline docs/privacy-api.md]:
```suggestion
function form_response_eraser( string $email, $page = 1 ) : array {
```

File: docs/privacy-api.md (hunk 1 above)
Comment (roborourke) [reply docs/privacy-api.md]:
```suggestion
Under the [Settings > Privacy admin menu item](internal://admin/privacy.php) it is possible to select which page you would like to use as the privacy policy page.
```

File: docs/privacy-api.md [hunk 11]
```diff
@@ -0,0 +1,259 @@
...
+		),
+		'https://policies.google.com/privacy?hl=en'
+	);
+	wp_add_privacy_policy_content( 'google-analytics', $policy_text );
+} );
+```
+
+## Personal Data Exports
+
+When a user makes a request for an export of their personal data a confirmation request should be sent to them via the [ Export Personal Data tool](internal://admin/tools.php?page=export_personal_data) by filling in their email address and clicking send.
```
Comment (roborourke) [inline docs/privacy-api.md]:
```suggestion
When a user makes a request for an export of their personal data a confirmation request should be sent to them via the [Export Personal Data tool](internal://admin/tools.php?page=export_personal_data) by filling in their email address and clicking send.
```

File: docs/privacy-api.md (hunk 4 above)
Comment (roborourke) [reply docs/privacy-api.md]: For the sake of example code it's probably ok but can do

File: docs/privacy-api.md [hunk 12]
```diff
@@ -0,0 +1,259 @@
...
+	return [
+		'data' => $export_items,
+		'done' => $responses->have_posts(),
+	];
+}
+```
+
+## Erasing Personal Data
+
+Following a similar mechanism to the personal data export should a user request that their personal data be deleted you can send a confirmation email to them via the CMS admin.
```
Comment (roborourke) [inline docs/privacy-api.md]:
```suggestion
Following a similar mechanism to the personal data export should a user request that their personal data be deleted you can send a confirmation email to them via the [Tools > Erase Personal Data page](internal://admin/tools.php?page=remove_personal_data).
```

**Comment by roborourke** [comment]
@joehoyle done :)

**Pull request merged**